run-tests.py

```diff
@@ -10,8 +10,9 @@
 import pybamm
 import sys
 import argparse
-import unittest
 import subprocess
+import pytest
+import unittest
```

Comment:
How much extra work would it be to remove unittest so it becomes pure pytest?

Reply:
It can be done in another ticket, I am just curious since right now we are using both libraries for testing

Reply:
I guess I can answer this a bit – @prady0t had previously mentioned that at least half of the integration tests fail currently with `pytest` – so an estimated timeline could be one chunk of the duration of a standard twelve-week GSoC project (plus or minus a few weeks for any associated technical or non-technical reasons). The tests will have to be rewritten with new syntax either manually or through an AST-based parsing tool.

Reply:
12 weeks sounds like a lot, why is it so high?

Reply:
We would need some other things from the project too. Some of them, off the top of my mind:

1.  Rewriting all the tests (unit too, besides integration) to pytest-style tests
2. Testing the built wheels (`pybamm.test()` similar to what is offered by other projects)
4. Improving test coverage with `hypothesis` (our code coverage is already high, but we could deal with a lot less verbosity when writing tests and check for any uncovered cases)
5. Adding pytest plugins based on ideas from other repositories

so all that does end up being a 175 hour project – and also, a potential mentees are mostly newcomers to open source software projects, so we need to give them time to explore the codebase and learn, so all that adds up

Reply:
We would need some other things from the project too. Some of them, off the top of my mind:

1.  Rewriting all the tests (unit too, besides integration) to pytest-style tests
2. Testing the built wheels (`pybamm.test()` similar to what is offered by other projects)
4. Improving test coverage with `hypothesis` (our code coverage is already high, but we could deal with a lot less verbosity when writing tests and check for any uncovered cases)
5. Adding pytest plugins based on ideas from other repositories

so all that does end up being a 175 hour project – and also, a potential mentees are mostly newcomers to open source software projects :), so we need to give them time to explore the codebase and learn, so all that adds up